OneCommunityGlobal/HGNRest
# Roberto -adds opencage http request to backend timezone route

# Description
Changing the OpenCage http request to happen on backend rather than client. Keeping the get request on the client allows for security risks since anyone can view the network tab and see the api key. By transitioning the request to happen on backend the api key can stay secure and concealed while just returning needed information to client.

## Related PRS (if any):
To test this backend PR you need to checkout the #[1640](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1640) frontend PR.


## Main changes explained:
- Updated TimeZoneAPIController to handle the opencage http request and return needed information to client
- Added Axios as a dependency to perform third party http request


## How to test:
1. check into current branch
2. sign up for an api key at `https://opencagedata.com` - it's free
3. add `TIMEZONE_PREMIUM_KEY=your api key` and 
`TIMEZONE_COMMON_KEY= your api key` to your environmental variables
4. do `npm install` and `npm run build && npm start` to run this PR locally
5. follow FE instructions


## Note:
Make sure to `npm install ` - a new dependency is added
Make sure to compile code after adding to your .env
